go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -157,7 +157,12 @@ def run_go_get(dependencies = [])
           tmp_go_file = "#{SecureRandom.hex}.go"
 
           package = Dir.glob("[^\._]*.go").any? do |path|
-            !File.read(path).include?("// +build")
+            # The build syntax changed from 1.16-1.18:
+            # https://go.dev/design/draft-gobuild
+            # https://go.dev/doc/go1.18#go-command
+            # old syntax: // +build
+            # new syntax: //go:build
+            !File.read(path).include?("/(\/\/ \+build)|(\/\/go:build)/")
```

Comment:
> I am wondering though, since we're still calling `File.read`, does this make a difference in terms of memory usage? You may have meant to write `File.open`.

Indeed I did. `File.open` is the right method here. Thanks for catching that.

Reply:
Thanks @mattt / @jurre for not only the code suggestions, but also the explanations.

Reply:
Shouldn't `^//\s*build` actually be `^//\s*\+build`? To capture the `+` in the old syntax?

Reply:
Oh silly me, this regex disappeared with the refactor to always write the temp file, so this doesn't even matter.